application: serial_lte_modem: Re-enable TFTP client (#10812)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/FTP_AT_commands.rst

::

   AT#XTFTP=<op>,<url>,<port>,<file_path>[,<mode>,<data>]

* The ``<op>`` parameter can accept one of the following values:

  * ``1`` - TFTP read request using IP protocol family version 4.
  * ``2`` - TFTP write request using IP protocol family version 4.
  * ``3`` - TFTP read request using IP protocol family version 6.
  * ``4`` - TFTP write request using IP protocol family version 6.

* The ``<url>`` parameter is a string.
  It indicates the hostname or the IP address of the TFTP server.
  Its maximum size is 128 bytes.
  When the parameter is an IP address, it supports both IPv4 and IPv6.
* The ``<port>`` parameter is an unsigned 16-bit integer (0 - 65535).

>   It represents the TFTP service port on the remote server, which is 69 normally.

Reviewer: This change makes it sound like there is no default value, and the user must instead apply the value that is normally 69. Is that the case?

Author: Yes no default value now

>   It represents the TFTP service port on the remote server, which is normally 69.

Reviewer: ```suggestion
  It represents the TFTP service port on the remote server, which is usually port 69.
```

* The ``<file_path>`` parameter is a string.
  It indicates the file path on the TFTP server to read from or write to.
  Its maximum size is 128 bytes.
* The ``<mode>`` parameter is a string.
  It indicates the three modes defined in TFTP protocol.
  Valid values are ``netascii``, ``octet`` and ``mail``.

>   The default value ``octet`` is applied if this parameter is omitted.

Author: I don't know why, just follow Zephyr TFTP lib.

>   The default value ``octet`` is applied if this parameter is omitted.

Reviewer 2: I guess the example: 'AT#XTFTP=2,"tftp-server.com",69,"test_put_01.txt","octet","test send TFTP PUT"' might be enough for this.

* The ``<data>`` parameter is a string.

>   When ``<op>`` is ``2`` or ``4``, this is the data to be put to remote server.

Reviewer: ```suggestion
  When ``<op>`` has a value of ``2`` or ``4``, this is the data to be put to the remote server.
```

   .. note::
      The maximum data size supported in WRITE request is 1024 bytes.

>       Based on protocol, TFTP connection is closed after writing last 512 (or less) bytes.

Reviewer: ```suggestion
      Based on protocol, TFTP connection is closed after writing last 512 (or less) bytes.
```
Can you explain what this means? I'm not sure I quite get it.

Reviewer: I'm guessing the termination is the more important part, so putting that in the front might be better:
```suggestion
      The TFTP connection is terminated by writing less than 512 bytes, as defined in the protocol specification (RFC 1350).
```
And if we're referring to the protocol, we should probably link to it as well here, adding it to the `link.txt`.

Author: Looks good. The RFC spec is that you can send as large as you like, in 512-byet chunks.
Once the last chunk, 0~511 bytes, is done, the session is over.
